extensions/packageManager/src/node/feeds/nuget/nugetFeed.ts

```diff
@@ -0,0 +1,127 @@
+        return this.asPackageDefinition(searchResult);
+      }
+    }
+
+    // Query package source url
+    // Example url that ends in this path: https://www.myget.org/F/ms-test-adapters/api/v3/index.json
+    let serviceIndex: INuGetServiceIndex = this.data as INuGetServiceIndex;
+
+    // If we don't have a service index provided at construction, get it from the url
+    if (!serviceIndex || !serviceIndex.resources) {
+      const httpResponse = await axios.get(this.packageSource.url);
+      serviceIndex = httpResponse?.data;
+    }
+
+    // Now that we have a service index either from seed data or from querying the NuGet feed,
+    // we get the query endpoint, build query parameters, call feed api and return query results to caller
+    const queryEndpoint = serviceIndex?.resources?.find((resource) => resource['@type'] === 'SearchQueryService');
+    if (queryEndpoint) {
+      const nuGetSearchUrl = this.buildNuGetSearchUrl(queryEndpoint['@id'], query);
+
+      this.composer.log(`Retrieving NuGet packages with computed search url ${nuGetSearchUrl}`);
+      const httpResponse = await axios.get(nuGetSearchUrl);
+
+      if (!httpResponse || !httpResponse.data) {
+        this.composer.log(`Failed to get search service from url ${nuGetSearchUrl}`);
+        return [];
+      }
+
+      const searchResult = httpResponse.data as INuGetSearchResult;
+      if (searchResult.data) {
+        return this.asPackageDefinition(searchResult);
+      }
+    } else {
+      this.composer.log(`NuGet index returned no query endpoints`);
+      return [];
+    }
+  }
+
+  /**
+   * Maps NuGet specific search result to the generic IPackageDefinition that is package source agnostic.
+   * @param searchResult The search result received from nuget.
+   */
+  private asPackageDefinition(searchResult: INuGetSearchResult): IPackageDefinition[] {
+    return searchResult.data.map((i) => {
+      return <IPackageDefinition>{
+        name: i.id,
+        version: i.version,
+        versions: i.versions ? semverSort.desc(i.versions.map((v) => v.version)) : [i.version],
+        authors: i.authors[0],
+        keywords: i.tags,
+        repository: i.projectUrl,
+        description: i.description,
+        language: this.language,
+        source: this.source,
+      };
+    });
+  }
+
+  /**
+   * Build NuGet search url based on the NuGet search spec.
+   * Spec: https://docs.microsoft.com/en-us/nuget/api/search-query-service-resource#search-for-packages
+   * @param baseUrl The NuGet search service url.
+   * @param query The desired query parameters.
+   */
+  private buildNuGetSearchUrl(baseUrl: string, query?: IPackageQuery): string {
+    // If a query is provided, it takes precedence. Otherwise, default to the package source one.
+    const effectiveQuery = query ?? this.packageSource?.defaultQuery;
+    let url: string = `${baseUrl}?prerelease=${effectiveQuery?.prerelease ?? true}&semVerLevel=${
+      effectiveQuery?.semVerLevel ?? '2.0.0'
```

Comment:
Why is 2.0.0 the default here?

Reply:
This is the default value we chose based on the docs. I can add a link to the spec here.